internal/cmd/template_pluginpreset.go

```diff
@@ -0,0 +1,283 @@
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if err := o.validate(); err != nil {
+				return err
+			}
+
+			if err := o.complete(); err != nil {
+				return err
+			}
+
+			return o.run()
+		},
+	}
+
+	cmd.Flags().StringVar(&o.pluginPresetPath, "pluginpreset", "", "Path to PluginPreset YAML file (required)")
+	cmd.Flags().StringVar(&o.pluginDefinitionPath, "plugindefinition", "", "Path to ClusterPluginDefinition YAML file (required)")
+	cmd.Flags().StringVar(&o.clusterName, "cluster", "", "Cluster name (required)")
+
+	if err := cmd.MarkFlagRequired("pluginpreset"); err != nil {
+		setupLog.Error(err, "Flag could not set as required", "pluginpreset")
+	}
+
+	if err := cmd.MarkFlagRequired("plugindefinition"); err != nil {
+		setupLog.Error(err, "Flag could not set as required", "plugindefinition")
+	}
+
+	if err := cmd.MarkFlagRequired("cluster"); err != nil {
+		setupLog.Error(err, "Flag could not set as required", "cluster")
+	}
+
+	return cmd
+}
+
+func (o *templatePluginPresetOptions) validate() error {
+	if _, err := os.Stat(o.pluginPresetPath); os.IsNotExist(err) {
+		return fmt.Errorf("PluginPreset file does not exist: %s", o.pluginPresetPath)
+	}
+
+	if _, err := os.Stat(o.pluginDefinitionPath); os.IsNotExist(err) {
+		return fmt.Errorf("ClusterPluginDefinition file does not exist: %s", o.pluginDefinitionPath)
+	}
+
+	return nil
+}
+
+func (o *templatePluginPresetOptions) complete() error {
+	var err error
+	o.clusterPluginDefinition, err = loadFromFile[greenhousev1alpha1.ClusterPluginDefinition](o.pluginDefinitionPath)
+	if err != nil {
+		return err
+	}
+
+	o.pluginPreset, err = loadFromFile[greenhousev1alpha1.PluginPreset](o.pluginPresetPath)
+	if err != nil {
+		return err
+	}
+
+	if err := o.validateCompatibility(); err != nil {
+		return err
+	}
+
+	if err := o.prepareValues(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (o *templatePluginPresetOptions) run() error {
+	helmValues, err := helminternal.ConvertFlatValuesToHelmValues(o.values)
+	if err != nil {
+		return err
+	}
+
+	valuesYAML, err := yaml.Marshal(helmValues)
+	if err != nil {
+		return err
+	}
+
+	f, err := os.CreateTemp("", "values-*.yaml")
+	if err != nil {
+		return err
+	}
+	defer func() {
+		f.Close()
+		os.Remove(f.Name())
+	}()
+
+	if _, err := f.Write(valuesYAML); err != nil {
+		return err
+	}
+
+	return o.runHelmTemplate(f.Name())
+}
+
+func (o *templatePluginPresetOptions) validateCompatibility() error {
+	expectedPluginDef := o.pluginPreset.Spec.Plugin.PluginDefinition
+	actualPluginDef := o.clusterPluginDefinition.Name
+
+	if expectedPluginDef != actualPluginDef {
+		return fmt.Errorf("PluginPreset references ClusterPluginDefinition '%s' but provided file defines '%s'", expectedPluginDef, actualPluginDef)
+	}
+
+	if o.clusterPluginDefinition.Spec.HelmChart == nil {
+		return fmt.Errorf("ClusterPluginDefinition '%s' must have a HelmChart reference", actualPluginDef)
+	}
```

Comment:
Here an additional check for the `kind`s of the PluginPreset & PluginDefinition should give an explicit error when wrong/incompatible files are passed to the command.